quiche/src/pmtud.rs

```diff
@@ -1,92 +1,193 @@
 #[derive(Default)]
 pub struct Pmtud {
-    /// The current path MTU estimate.
-    cur_size: usize,
+    /// The current PMTU estimate.
+    current_mtu: usize,
 
-    /// The last MTU probe size that was attempted.
-    probe: usize,
+    /// The last PMTU probe size that was attempted.
+    probe_size: usize,
 
-    /// Indicated if Path MTU probe needs to be generated.
-    next_size: bool,
+    /// Whether or not a PMTU probe needs to be generated.
+    should_probe: bool,
 
-    /// Check config for PMTU variable.
-    enable: bool,
+    /// Whether or not PMTUD is enabled.
+    enabled: bool,
 }
 
 impl Pmtud {
     /// Creates new PMTUD instance.
-    pub fn new(cur_size: usize) -> Self {
+    pub fn new(initial_mtu: usize) -> Self {
         Self {
-            cur_size,
+            current_mtu: initial_mtu,
```

Comment:
We should think about the invariants that this code should have.  Some possible examples:

if `!enabled`, then `!should_probe`.  Also known as `assert!(enabled || !should_probe)`.

If `should_probe`, `current_mtu <= probe_size`.  Or even `current_mtu < probe_size` since probing only makes sense if the probe size is larger than the MTU in use. This one makes me think that we should initialize `probe_size = initial_mtu` or have the setter for `should_probe` take `probe_size` as an argument: `disable_probe()` and `schedule_probe(size)`

And `initial_mtu >= 1200` since that's required by QUIC.

Reply:
I agree with all of your points, though I think those will be best suited for a followup PR. I added the assertion that `initial_mtu >= 1200` for now